agent/engine/docker_task_engine.go

```diff
@@ -873,6 +873,12 @@ func (engine *DockerTaskEngine) pullAndUpdateContainerReference(task *apitask.Ta
 		return metadata
 	}
 	pullSucceeded := metadata.Error == nil
+	if pullSucceeded && engine.cfg.ContainerPullInParallel.Enabled() {
```

Comment:
Do you need to check for this config var here since all pulled containers should be added to state?

Reply:
Removed the check `engine.cfg.ContainerPullInParallel.Enabled()` from here. Thanks!